docs/src/content/docs/drivers/postgres/getting-started.mdx

```diff
@@ -0,0 +1,216 @@
+---
+title: Postgres Drivers and Compatibility
+---
+
+import Fiddle from '@components/fiddle.astro';
+import Txs from '@components/fiddle/txs.astro';
+import Query from '@components/fiddle/query.astro';
+import { Aside } from '@astrojs/starlight/components';
+import { Card, CardGrid } from '@astrojs/starlight/components';
+import TabCssFix from '@components/tabcssfix.astro';
+import { Tabs, TabItem } from '@astrojs/starlight/components';
+
+<TabCssFix></TabCssFix>
+
+XTDB supports the Postgres wire protocol (often referred to as 'PGWire') as the primary API interface.
+
+Whilst XTDB offers some level of compatibility with Postgres' actual SQL dialect, there are many syntactic and functional capabilities which Postgres offers that are not supported in XTDB, and users should generally always refer to XTDB's [SQL documentation](/reference/main/sql/queries).
+
+The advantage of embracing wire protocol compatibility is that many PostgreSQL clients or drivers are able to connect to XTDB seamlessly and run many useful queries without issue. There will always be exceptions and this page will document them as comprehensively as possible.
+
+## Usage examples
+
+<div data-no-swup>
+  <Tabs>
+    <TabItem label="Node.js">
+This example uses the [`postgres`](https://www.npmjs.com/package/postgres) package.
+
+```javascript
+"use strict"
+
+import postgres from 'postgres';
+
+const sql = postgres({
+  host: "localhost",
+  port: 5432,
+  fetch_types: false, // currently required to be 'false'
+  // debug: console.log
+});
+
+// Mapping of domain JavaScript types to Postgres OIDs
+const typeToOid = {
+  number: 23,  // Postgres OID for 'int4' (note that 20/'int8' will coerce to string by default due JS 53-bit numeric precision)
+  string: 25   // Postgres OID for 'text
+};
+// for the full list of OIDs see "Supported types" for the Postgres Driver at https://docs.xtdb.com
+
+function objsToTypedObjs(objs) {
+  return objs.map(o => {
+    const transformedEntry = {};
+    for (const key in o) {
+      if (o.hasOwnProperty(key)) {
+        const jsType = typeof o[key];
+        const oid = typeToOid[jsType];
+        if (oid) {
+          transformedEntry[key] = sql.typed(o[key], oid);
+        } else {
+          throw new Error(`Unsupported JS type: ${jsType} for key: ${key}`);
+        }
+      }
+    }
+    return transformedEntry;
+  });
+}
```

Comment:
Do we have a ticket to link to for getting rid of this?

Reply:
@jarohen took a note earlier to open one to the effect of "let's ship a small library" - but since I'm here I just opened https://github.com/xtdb/xtdb/issues/3606 - will link to it :+1: